File: bob/bob.rb
# lackadaisical bob
class Bob
  def self.hey(remark)
    if remark.match(/\p{Lower}/).nil? && remark.match(/[a-zA-Z]/)
      'Whoa, chill out!'
    elsif remark.strip.empty?
      'Fine. Be that way!'
    elsif remark.strip.split('').last.=~(/\?/)
      'Sure.'
    else
      'Whatever.'
    end
  end
end

module BookKeeping
  VERSION = 1
end

File: javascript/bob/bob.js
(function() {
  var Bob;

  Bob = class Bob {
    hey(phrase) {
      if (!phrase.match(/[a-z]/) && phrase.match(/[A-Z]/)) {
        return 'Whoa, chill out!';
      } else if (phrase.trim() === "") {
        return 'Fine. Be that way!';
      } else if (phrase.trim().endsWith('?')) {
        return 'Sure.';
      } else {
        return 'Whatever.';
      }
    }

  };

  module.exports = Bob;

}).call(this);
